source/mergeSpec.js

```diff
@@ -0,0 +1,68 @@
+import _isObject from './internal/_isObject';
+import _curry1 from './internal/_curry1';
+import apply from './apply';
+import curryN from './curryN';
+import map from './map';
+import max from './max';
+import merge from './merge';
+import pluck from './pluck';
+import reduce from './reduce';
+import values from './values';
+
+/**
+ * Given a spec object recursively mapping properties to functions, creates a
+ * function producing an object of the same structure, by mapping each property
+ * to the result of calling its associated function with the supplied arguments.
+ * Nearly identical to [applySpec()](http://ramdajs.com/docs/#applySpec) with
+ * one difference being it passes through any props unaltered on the final object
+ * for which no "spec" functions were defined (similar to how
+ * [evolve()](https://ramdajs.com/docs/#evolve) passes through any props unaltered
+ * that you did not name in the spec).
+ *
+ * @func
+ * @memberOf R
+ * @since v0.25.1
+ * @category Function
+ * @sig {k: ((a, b, ..., m) -> v)} -> ((a, b, ..., m) -> {k: v})
+ * @param {Object} spec An object recursively mapping properties to functions for
+ *        producing values for these properties
+ * @return {Function} A function that returns an object of the same structure
+ * as `spec`, with each property set to the value returned by calling its
+ * associated function with the supplied arguments.
+ * @see R.applySpec, R.converge, R.juxt
+ * @example
+ *
+ *      const formatUser = R.mergeSpec({
+ *        fullName: R.compose(R.join(' '), R.values, R.pick(['firstName', 'lastName'])),
+ *        address: R.evolve({
+ *          street: R.trim,
+ *          city: compose(str => str.replace(/(?:^|\s)\S/g, R.toUpper), R.trim),
+ *          state: R.toUpper,
+ *          zip: compose(R.trim, R.when(R.is(Number), R.toString))
+ *        })
+ *      });
+ *
+ *      formatUser({
+ *        firstName: 'Montgomery',
+ *        lastName: 'Burns',
+ *        address: {
+ *          street: '1000 Mammon Lane, ',
+ *          city: 'springfield',
+ *          state: 'or',
+ *          zip: 97403
+ *        }
+ *      }); // => { firsName: 'Montgomery', lastName: 'Burns', fullName: 'Montgomery Burns', address: { street: '1000 Mammon Lane,', city: 'Springfield', state: 'OR', zip: '97403' } }
+ */
+var mergeSpec = _curry1(function mergeSpec(spec) {
+  spec = map(function(v) { return typeof v == 'function' ? v : mergeSpec(v); },
+             spec);
+  return curryN(reduce(max, 0, pluck('length', values(spec))),
+                function() {
+                  var args = arguments;
+                  if (args.length === 1 && _isObject(args[0])) {
+                    return merge(args[0], map(function(f) { return apply(f, args); }, spec));
+                  }
+                  return map(function(f) { return apply(f, args); }, spec);
+                });
+});
```

Comment:
Something is wrong with the indentation here.

Reply:
I just tried to figure out what it might be (there weren't any eslint alarms in the final version of the file, but I agree it looks a little weird). I had the `source/applySpec.js` open when creating this file, to use as a guide, so perhaps I shouldn't have been following that one? The formatting `source/anyPass.js` looks a little more like what I should probably follow.

Also, I just realized that a lot of the source code in Ramda line breaks before reaching 80 character on a single line. I can change line 63 to observe that convention too.

Reply:
Maybe more like this?

```
var mergeSpec = _curry1(function mergeSpec(spec) {
  spec = map(function(v) { return typeof v == 'function' ? v : mergeSpec(v); },
             spec);
  return curryN(reduce(max, 0, pluck('length', values(spec))), function() {
    var args = arguments;
    if (args.length === 1 && _isObject(args[0])) {
      return merge(args[0], map(function(f) { return apply(f, args); }, spec));
    }
    return map(function(f) { return apply(f, args); }, spec);
  });
});
```